ra/ra.go

```diff
@@ -333,36 +334,70 @@ func (ra *RegistrationAuthorityImpl) checkRegistrationLimits(ctx context.Context
 }
 
 // NewRegistration constructs a new Registration from a request.
-func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, init core.Registration) (core.Registration, error) {
-	if err := ra.keyPolicy.GoodKey(ctx, init.Key.Key); err != nil {
-		return core.Registration{}, berrors.MalformedError("invalid public key: %s", err.Error())
+func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, request *corepb.Registration) (*corepb.Registration, error) {
+	// Func to return false if the key length or ip are zero
+	newRegistrationValid := func(reg *corepb.Registration) bool {
```

Comment:
Since you're only calling this once, let's not make it a function. Instead, let's put its checks into the if below:

`if request == nil || len(reg.Key) == 0 || len(reg.InitialIP) == 0`